### PR TITLE
Add EventType and EventSource for Edge Media

### PR DESCRIPTION
**Pull request by kevinlind**
## Description

Adding EventType and EventSource for Edge Media.

See also Android Core PR with similar change: https://github.com/adobe/aepsdk-core-android/pull/403

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Review by addb — APPROVED** [review]
looks good to me!

File: AEPCore/Sources/eventhub/EventSource.swift [hunk 1]
```diff
@@ -36,4 +36,8 @@ public class EventSource: NSObject {
     public static let applicationClose = "com.adobe.eventSource.applicationClose"
     public static let personalizationDecisions = "personalization:decisions"
     public static let locationHintResult = "locationHint:result"
+    public static let errorResponseContent = "com.adobe.eventSource.errorResponseContent";
+    public static let requestTracker = "com.adobe.eventSource.requestTracker";
+    public static let trackMedia = "com.adobe.eventSource.trackMedia";
+    public static let sessionCreated = "com.adobe.eventSource.sessionCreated";
```
Comment (praveek) [inline AEPCore/Sources/eventhub/EventSource.swift:42]: We previously dispatched a similar event in media to match client API events with hitdebugger response. Does this still apply to edge media?

File: AEPCore/Sources/eventhub/EventSource.swift [hunk 2]
```diff
@@ -36,4 +36,8 @@ public class EventSource: NSObject {
     public static let applicationClose = "com.adobe.eventSource.applicationClose"
     public static let personalizationDecisions = "personalization:decisions"
     public static let locationHintResult = "locationHint:result"
+    public static let errorResponseContent = "com.adobe.eventSource.errorResponseContent";
+    public static let requestTracker = "com.adobe.eventSource.requestTracker";
```
Comment (praveek) [inline AEPCore/Sources/eventhub/EventSource.swift:40]: nit: Since media doesn't send a corresponding response event, should we use createTracker instead?

**Comment by praveek** [comment]
Are there plans to update media direct extension to use the new event type/source?

**Comment by kevinlind** [comment]
> Are there plans to update media direct extension to use the new eventsource?

We've previously argued that event are considered part of the public API. The media direct extensions use a naming convention which included the extension name in the event source string ("com.adobe.eventsource.media.trackmedia"). I would like to fix that with this PR, but then we would not be able to update the media direct extensions without the risk of breaking someone's implementation if they are listening for these events.

File: AEPCore/Sources/eventhub/EventSource.swift (hunk 1 above)
Comment (addb) [reply AEPCore/Sources/eventhub/EventSource.swift:42]: We have kept this for now. We don't have the integration with hitdebugger for now, but I am thinking we might need this for low level events validation in griffon.

**Comment by emdobrin** [comment]
I agree with @kevinlind that we should not modify the event types/sources for Media direct as it may be breaking some implementations.

File: AEPCore/Sources/eventhub/EventSource.swift (hunk 1 above)
Comment (emdobrin) [reply AEPCore/Sources/eventhub/EventSource.swift:42]: @addb there will likely be no client-side integration with hitdebugger in the case of Edge Media. For debugging purposes can we relay exclusively on the `media-analytics:new-session` response handle and not dispatch this additional internal event?

**Comment by kevinlind** [comment]
Closing PR. We'll keep the type/source internal to Edge Media for now and reevaluate the need for this PR closer to the Edge Media GM.

**Pull request closed without merging**